### PR TITLE
build: wifi: allow espressif boards to test net/wifi use cases

### PR DESCRIPTION
**Pull request by sylvioalves**
This aims to allow Espressif boards to test Wi-Fi drivers without binary blobs. The following changes were added:

1) Convert `NRF_WIFI_BUILD_ONLY_MODE` and `NXP_WIFI_BUILD_ONLY_MODE` to use the same common Kconfig name `BUILD_ONLY_NO_BLOBS`.
2) Modify `hal_nxp` and `hal_espressif` to support the above change. Include espressif Wi-Fi RF stubs.
3) Modify all esp32-based boards.yml to allow `net` tests.

### CODE REVIEW

**Comment by sylvioalves** [comment]
@krish2718 @dleach02 As you can see in this PR, I have aimed to consolidate the `WIFI_BUILD_ONLY_MODE` as common Kconfig. I have also submitted a similar approach to BT driver as you can see in #80786.  

Please, let me know your thoughts about both. I think that we could rename `XXX_BUILD_ONLY_MODE` to something more clear as such as `XXX_BUILD_WITH_NO_BLOBS` os similar. 

1) Are you OK making this symbol global?
2) Does it sound good renaming it?
3) We could create a global symbol that could be used for both Wi-Fi and BT instead of splitting in 2. Any thoughts? Could b something like `CONFIG_BUILD_WITH_NO_BLOBS`. 

@carlescufi PTAL.

**Review by krish2718 — DISMISSED** [review]
LGTM, definitely makes sense to move this to global as it's a common problem.

File: drivers/wifi/nrfwifi/CMakeLists.txt
Comment (krish2718) [inline drivers/wifi/nrfwifi/CMakeLists.txt:158]: this should now be `if(NOT CONFIG_WIFI_BUILD_ONLY_MODE)`

File: samples/net/wifi/shell/sample.yaml
Comment (kartben) [inline samples/net/wifi/shell/sample.yaml:91]: consider adding esp32c6 given https://github.com/zephyrproject-rtos/zephyr/pull/81078 might get merged by the time you will be coming back to this PR? :)

**Review by kartben — DISMISSED** [review]
@sylvioalves approved in principle - needs rebasing though!

**Review by wmrsouza — DISMISSED** [review]

File: samples/net/wifi/shell/sample.yaml
Comment (sylvioalves) [reply samples/net/wifi/shell/sample.yaml:91]: Sure, will rebase and add C6 accordingly.

**Review by kartben — COMMENTED** [review]
Apologies for apparently missing this in my earlier review but can you please document the renaming of the Kconfig options in the migration guide? This might impact downstream users and their out-of-tree boards

**Comment by sylvioalves** [comment]
> Apologies for apparently missing this in my earlier review but can you please document the renaming of the Kconfig options in the migration guide? This might impact downstream users and their out-of-tree boards

Done!

**Review by kartben — DISMISSED** [review]

**Review by jukkar — COMMENTED** [review]
One of the commit messages says:
```
This lets esp32-based boards to be tested regarding
Wi-Fi changes. WIP
```
About the WIP, wondering now that is there more to come to that commit or in some subsequent commits, or was it just a left over from some earlier work?

**Review by wmrsouza — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by sylvioalves** [comment]
@dleach02 As you requested, pinging you so we can mode with hal_nxp.

**Comment by sylvioalves** [comment]
@krish2718 @jhedberg @jukkar as you can see in this PR and at #80786, we could end up having 2 different Kconfig to enable CI build without blobs (for Wi-Fi and BT). We could convert both to a single Kconfig entry as discussed in previous comments, i.e., `CONFIG_BUILD_ONLY_NO_BLOBS`. Let me know if you are ok with it and I shall update the PRs accordingly.

**Comment by krish2718** [comment]
Agreed, this should be a common option for builds that use blobs.

**Comment by jhedberg** [comment]
@sylvioalves Thanks for taking this initiative! It sounds like a quite useful feature, and something we'll certainly want to support for Silabs as well. The single generic Kconfig option seems like the best approach to me.

**Comment by sylvioalves** [comment]
PR was updated with the common `CONFIG_BUILD_ONLY_NO_BLOBS`. Will merge hal side when we get proper approvals here. Thanks.

**Review by krish2718 — DISMISSED** [review]

File: drivers/wifi/Kconfig
Comment (jhedberg) [inline drivers/wifi/Kconfig:38]: I guess `drivers/wifi/Kconfig` is the wrong place for this, now that it's completely generic. Perhaps `modules/Kconfig` instead?

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: drivers/wifi/Kconfig
Comment (jhedberg) [reply drivers/wifi/Kconfig:38]: `BUILD_WITHOUT_BLOBS` is fine

Comment (krish2718) [reply drivers/wifi/Kconfig:38]: `BUILD_WITHOUT_BLOBS` is fine, `ONLY` was added to imply that this is `build-only` and cannot be used on a real board for testing.

Comment (jhedberg) [reply drivers/wifi/Kconfig:38]: > `BUILD_WITHOUT_BLOBS` is fine, `ONLY` was added to imply that this is `build-only` and cannot be used on a real board for testing.

Yeah, that was my idea when I proposed it, but then I realized that when you try to read it as normal English it's mildly confusing. Either way is fine with me.

Comment (jhedberg) [reply drivers/wifi/Kconfig:38]: Also, as long as we output a clear warning during build, I think that should do the job of informing the user that the result is something non-functional.

**Review by jukkar — DISMISSED** [review]

**Review by wmrsouza — DISMISSED** [review]

**Comment by jhedberg** [comment]
@sylvioalves some rebasing is needed. I'd be happy to approve once the Kconfig option has been moved away from its current wifi-specific location.

**Comment by sylvioalves** [comment]
> @sylvioalves some rebasing is needed. I'd be happy to approve once the Kconfig option has been moved away from its current wifi-specific location.

I was expecting `hal_nxp` to be merged so that I could update this PR accordingly.

**Comment by dleach02** [comment]
> > @sylvioalves some rebasing is needed. I'd be happy to approve once the Kconfig option has been moved away from its current wifi-specific location.
> 
> I was expecting `hal_nxp` to be merged so that I could update this PR accordingly. Will do so.

This PR is tricky because we don't want to merge the NXP HAL until this one is fully ready to merge. There are a lot of NXP HAL PRs that need to be synchronized. The ESP hal PR has merged so I would update this PR with the SHA. 

@jukkar seems to have one request yet on a kconfig change, has that been done?

**Review by jhedberg — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by sylvioalves** [comment]
@dleach02 I've rebased `hal_nxp` PR. Will rebase this one and submit again once merged in there. Up to you now.

**Comment by sylvioalves** [comment]
@dleach02, are you able to move on with this? Do you expect anything else from my side?

File: doc/releases/migration-guide-4.1.rst
Comment (dleach02) [inline doc/releases/migration-guide-4.1.rst:220]: The NXP option being changed is CONFIG_NXP_WIFI_BUILD_ONLY_MODE

**Comment by dleach02** [comment]
@sylvioalves do you know what is causing the CI failures? 

Because we have a lot of NXP HAL side PRs we don't merge them until the Zephyr side is completely ready to be merged. Otherwise we end up blocking things that are ready.

**Comment by dleach02** [comment]
The rebase passes my internal tests.

**Comment by sylvioalves** [comment]
PR is ready now.

**Review by kartben — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
